src/OSD.cpp

```diff
@@ -814,88 +814,43 @@ int OSD::start()
     return ret;
 }
 
-int OSD::exit()
+void OSD::cleanup_item(OSDItem *osdItem)
 {
     int ret;
 
-    ret = IMP_OSD_Stop(osdGrp);
-    LOG_DEBUG_OR_ERROR(ret, "IMP_OSD_Stop(" << osdGrp << ")");
-
-    if (osdPrivBack.imp_rgn)
-    {
-        ret = IMP_OSD_ShowRgn(osdPrivBack.imp_rgn, osdGrp, 0);
-        LOG_DEBUG_OR_ERROR(ret, "IMP_OSD_ShowRgn(osdPrivBack.imp_rgn, " << osdGrp << ", 0)");
-    }
-
-    if (osdPrivText.imp_rgn)
-    {
-        ret = IMP_OSD_ShowRgn(osdPrivText.imp_rgn, osdGrp, 0);
-        LOG_DEBUG_OR_ERROR(ret, "IMP_OSD_ShowRgn(osdPrivText.imp_rgn, " << osdGrp << ", 0)");
-    }
-
-    ret = IMP_OSD_ShowRgn(osdTime.imp_rgn, osdGrp, 0);
-    LOG_DEBUG_OR_ERROR(ret, "IMP_OSD_ShowRgn(osdTime.imp_rgn, " << osdGrp << ", 0)");
-
-    ret = IMP_OSD_ShowRgn(osdUser.imp_rgn, osdGrp, 0);
-    LOG_DEBUG_OR_ERROR(ret, "IMP_OSD_ShowRgn(osdUser.imp_rgn, " << osdGrp << ", 0)");
-
-    ret = IMP_OSD_ShowRgn(osdUptm.imp_rgn, osdGrp, 0);
-    LOG_DEBUG_OR_ERROR(ret, "IMP_OSD_ShowRgn(osdUptm.imp_rgn, " << osdGrp << ", 0)");
-
-    ret = IMP_OSD_ShowRgn(osdLogo.imp_rgn, osdGrp, 0);
-    LOG_DEBUG_OR_ERROR(ret, "IMP_OSD_ShowRgn(osdLogo.imp_rgn, " << osdGrp << ", 0)");
-
-    if (osdPrivBack.imp_rgn)
-    {
-        ret = IMP_OSD_UnRegisterRgn(osdPrivBack.imp_rgn, osdGrp);
-        LOG_DEBUG_OR_ERROR(ret, "IMP_OSD_UnRegisterRgn(osdPrivBack.imp_rgn, " << osdGrp << ")");
-    }
-
-    if (osdPrivText.imp_rgn)
-    {
-        ret = IMP_OSD_UnRegisterRgn(osdPrivText.imp_rgn, osdGrp);
-        LOG_DEBUG_OR_ERROR(ret, "IMP_OSD_UnRegisterRgn(osdPrivText.imp_rgn, " << osdGrp << ")");
-    }
+    if (osdItem->imp_rgn == 0)
```

Comment:
Here I introduced another bug.. 0 is a valid region.. It is the first item created.. Since it is not deleted, an overlay image appears..
<img width="187" height="36" alt="imagen" src="https://github.com/user-attachments/assets/0006697a-7c4b-4208-ab0e-db0b608ae3ec" />
😞